### PR TITLE
Create documentation for challenge 3

### PR DESCRIPTION
**Pull request by davidkpiano**
This PR adds documentation for challenge 3.

### CODE REVIEW

File: docs/challenge-003.md [hunk 1]
```diff
@@ -2,8 +2,29 @@
 
 ## Challenge Objective
 
+For the last challenge, we want to write the aggregations from [Challenge 2](./challenge-002.md) into some "master data collection" (MDC). Again, we'll use a Cosmos DB collection as the MDC. After each store entity is finished aggregating the result, each item's aggregation should be written as a row in the MDC.
```
Comment (jcocchi) [inline docs/challenge-003.md]:
```suggestion
For the last challenge, we want to write the aggregations from [Challenge 2](./challenge-002.md) into some "master data collection" (MDC). Again, we'll use a Cosmos DB collection as the MDC. After each store entity is finished aggregating the result, each item's aggregation should be written as either a new document in the MDC (for the first time we see that item) or as an update to an existing document.
```

File: docs/challenge-003.md [hunk 2]
```diff
@@ -2,8 +2,29 @@
...
+
+Ultimately, this MDC will be used to quickly retrieve aggregations via some sort of API (which is beyond the scope of this challenge). The objective is to make sure the MDC is kept up-to-date with accurate inventory data.
+
 ## Brief Guidance
 
+This challenge will focus on the red outlined area of the below design:
+
+![Design](https://user-images.githubusercontent.com/1093738/82636141-ff2e9580-9bcf-11ea-9f84-38df789c088a.png)
```
Comment (jcocchi) [inline docs/challenge-003.md]: There is a red box in the key for "Azure Durable Functions" but I don't see any red in the diagram. Should the orchestrator/ entity/ activity functions be red instead?

File: docs/challenge-003.md [hunk 3]
```diff
@@ -2,8 +2,29 @@
...
+- [Azure Cosmos DB .NET Client](https://docs.microsoft.com/en-us/dotnet/api/overview/azure/cosmosdb?view=azure-dotnet)
+- [Azure Activity Functions](https://docs.microsoft.com/en-us/azure/azure-functions/durable/durable-functions-types-features-overview#activity-functions)
+- [Orchestrator function code constraints](https://docs.microsoft.com/en-us/azure/azure-functions/durable/durable-functions-code-constraints) (don't write directly to Cosmos DB from the orchestrator!)
+- [Accessing environment variables](https://docs.microsoft.com/en-us/sandbox/functions-recipes/environment-variables?tabs=csharp#accessing-environment-variables) in Azure Functions
+
 ## Challenge Completion Criteria
+
+To successfully complete the challenge:
+
+- [ ] Ensure that sending a "shipment" or "onHand" event writes the updated inventory data for the item as a new row in the MDC
```
Comment (jcocchi) [inline docs/challenge-003.md]:
```suggestion
- [ ] Ensure that sending a "shipment" or "onHand" event writes the updated inventory data for the item in the MDC
```

**Review by jcocchi — CHANGES_REQUESTED** [review]
Only the first and last comment should be considered blocking, the comment about the diagram is just a question

**Review by jcocchi — APPROVED** [review]
looks good!

File: docs/challenge-003.md [hunk 4]
```diff
@@ -2,8 +2,29 @@
...
+
+Ultimately, this MDC will be used to quickly retrieve aggregations via some sort of API (which is beyond the scope of this challenge). The objective is to make sure the MDC is kept up-to-date with accurate inventory data.
+
 ## Brief Guidance
 
+This challenge will focus on the red outlined area of the below design:
+
+![Architecture](https://user-images.githubusercontent.com/1093738/82704038-b9f77b80-9c42-11ea-8f64-a88fba66147c.jpeg)
+
+To accomplish this challenge, you will use the `mdc` container that you created in [Challenge 1](./challenge-001.md) to store inventory data that the orchestrator receives after calling an entity to perform some aggregation.
```
Comment (codepossible) [inline docs/challenge-003.md:15]: I think we need to provide some logic somewhere in the challenges on how the change feed need to map to updates to the master inventory. Should it be in the solution design document? This document is our last opportunity to provide that guidance.

File: docs/challenge-003.md [hunk 5]
```diff
@@ -2,8 +2,29 @@
 
 ## Challenge Objective
 
+For the last challenge, we want to write the aggregations from [Challenge 2](./challenge-002.md) into some "master data collection" (MDC). Again, we'll use a Cosmos DB collection as the MDC. After each store entity is finished aggregating the result, each item's aggregation should be written as either a new document in the MDC (for the first time we see that item) or as an update to an existing document.
+
+Ultimately, this MDC will be used to quickly retrieve aggregations via some sort of API (which is beyond the scope of this challenge). The objective is to make sure the MDC is kept up-to-date with accurate inventory data.
+
 ## Brief Guidance
```
Comment (codepossible) [inline docs/challenge-003.md]: We can drop the word "brief" here.

**Review by codepossible — APPROVED** [review]
Change MDC to MDS... 🐙

File: docs/challenge-003.md (hunk 4 above)
Comment (davidkpiano) [reply docs/challenge-003.md:15]: We'll do it in the solution design document.

File: docs/challenge-003.md (hunk 5 above)
Comment (davidkpiano) [reply docs/challenge-003.md]:
```suggestion
## Guidance
```

**Pull request merged**